view/frontend/web/instantsearch.js

```diff
@@ -420,48 +420,74 @@ requirejs(['algoliaBundle', 'Magento_Catalog/js/price-utils'], function (algolia
+                                    if (value && data) {
+                                        data.find((node) => {
+                                                if (node[key] === value) {
+                                                    tempObj.found = node;
+                                                    return node;
+                                                }
+                                                return filterCategories(node[sub], value, key, sub, tempObj);
+                                            }
+                                        );
+                                        if (tempObj.found) {
+                                            return tempObj.found;
+                                        }
+                                    }
+                                    return false;
+                                }
+                            ;
+                            if (algoliaConfig.request.childrenCategories) {
+                                items = filterCategories(items, activeCategory, 'label', 'data').data ?? [];
+                            } else {
+                                items = [];
+                            }
+                            return items.map(item=>addCategoryUrl(item));
+                        }else{
+                            return items.map(item => ({
+                                ...item,
+                                label: item.label,
+                            }));
                         }
-                        return items.map(item => ({
-                            ...item,
-                            label: item.label,
-                        }));
                     },
-				};
-
-				hierarchicalMenuParams.templates.item = '' +
-					'<a class="{{cssClasses.link}} {{#isRefined}}{{cssClasses.link}}--selected{{/isRefined}}" href="{{url}}">{{label}}' + ' ' +
-					'<span class="{{cssClasses.count}}">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span>' +
-					'</a>';
-				hierarchicalMenuParams.panelOptions = {
-					templates: {
-						header: '<div class="name">' + (facet.label ? facet.label : facet.attribute) + '</div>',
-					}
-				};
+                };
+
+                // hierarchicalMenuParams.templates.item = '' +
+                // 	'<a class="{{cssClasses.link}} {{#isRefined}}{{cssClasses.link}}--selected{{/isRefined}}" href="{{url}}">{{label}}' + ' ' +
+                // 	'<span class="{{cssClasses.count}}">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span>' +
+                // 	'</a>';
+                hierarchicalMenuParams.templates.item = '' + '<a class="{{cssClasses.link}} {{#isRefined}}{{cssClasses.link}}--selected{{/isRefined}}"' + 'href="{{categoryUrl}}">{{label}}' + ' ' + '<span class="{{cssClasses.count}}">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span>' + ' ' + '</a>';
```

Comment:
Setting the URL via template is insufficient. We need to address the event handler in the underlying InstantSearch lib. For this a custom renderer will be needed via a [connector](https://www.algolia.com/doc/api-reference/widgets/hierarchical-menu/js/#connector).